src/wrappers/OsipiBase.py

```diff
@@ -0,0 +1,210 @@
+import numpy as np
+import importlib
+from scipy.stats import norm
+
+
+class OsipiBase:
+    """The base class for OSIPI IVIM fitting"""
+    
+    def __init__(self, bvalues=None, thresholds=None, bounds=None, initial_guess=None, algorithm=None):
+        # Define the attributes as numpy arrays only if they are not None
+        self.bvalues = np.asarray(bvalues) if bvalues is not None else None
+        self.thresholds = np.asarray(thresholds) if thresholds is not None else None
+        self.bounds = np.asarray(bounds) if bounds is not None else None
+        self.initial_guess = np.asarray(initial_guess) if initial_guess is not None else None
+        
+        # If the user inputs an algorithm to OsipiBase, it is intereprete as initiating
+        # an algorithm object with that name.
+        if algorithm:
+            self.osipi_initiate_algorithm(algorithm, bvalues=bvalues, thresholds=thresholds, bounds=bounds, initial_guess=initial_guess)
+            
+    def osipi_initiate_algorithm(self, algorithm, **kwargs):
+        """Turns the class into a specified one by the input.
+        This method can be used instead of specifically importing that class and initiating it.
+        
+        WIP: Add args and kwargs!
+
+        Args:
+            algorithm (string): The name of the algorithm, should be the same as the file in the src/standardized folder without the .py extension.
+        """
+        
+        # Import the algorithm
+        import_base_path = "src.standardized"
```

Comment:
Does this always work or only when the current working directory is equal to the root dir of this repo? Maybe we should check if the root dir is added to pythonpath then?

Reply:
Not sure, haven't tested that. I'm always in the root directory of the folder. I could see if I can add a check for that. Do you have an idea of what we could do if it is not in pythonpath? Or do we just raise an error, asking people to be in the root dir?

Reply:
What is the advantage of this dynamic loading vs hard coding all the available algorithms into a file somewhere (or even this file) and referencing it that way?

Reply:
Manual work :) Just the possibility of forgetting to add one really. We already need to implement them according to our standard and in addition to that, add them to a registry of submissions. A third place/thing to do seems excessive. However, depending on how we do the registry, we could also just get the names from there. Of course, this is me assuming that we will drown in submissions in the future. And the chance of that happening seems quite slim at the moment.

Reply:
I've added a check to see if the root dir is in pythonpath or not. Unfortunately I don't have the time to find a better solution right now, but will attempt to for the next version that includes proper checks. If this is the only thing holding up the PR, I think we should let it pass so we can continue the work on testing

Reply:
I think the failing tests is the remaining thing. Easiest approach is to not use the ETP method. Perhaps I could work on it later to reintegrate it.